assets/blocks/course-outline/use-block-creator.js

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+const blockNames = {
+	module: 'sensei-lms/course-outline-module',
+	lesson: 'sensei-lms/course-outline-lesson',
+};
+
+/**
+ * Blocks creator hook.
+ * It adds blocks dynamically to the InnerBlock.
+ *
+ * @param {Object[]} blocksData Blocks data to insert.
+ * @param {string}   clientId   Block client ID.
+ */
+const useBlocksCreator = ( blocksData, clientId ) => {
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+
+	useEffect( () => {
+		if ( ! blocksData || 0 === blocksData.length ) {
+			return;
+		}
+
+		const blocks = blocksData.map( ( { type, ...block } ) =>
+			createBlock( blockNames[ type ], block )
+		);
+
+		replaceInnerBlocks( clientId, blocks, false );
```

Comment:
Might not be better here, just noting that Innerblocks have a controlled mode, so instead of calling replaceInnerBlocks, it could be `<InnerBlocks value={ blocks } onChange= { ... } />`

Reply:
Humm! Good to know! Thank you for that! 🥳 TIL
But yeah, I think the `replaceInnerBlocks` will be the cleaner way for this case.